spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactory.java

```diff
@@ -137,6 +150,9 @@ public DeadLetterPublishingRecoverer create() {
 		};
 
 		recoverer.setHeadersFunction((consumerRecord, e) -> addHeaders(consumerRecord, e, getAttempts(consumerRecord)));
+		if (this.headersFunction != null) {
```

Comment:
Yes; I was going to do it here too, but I had already added to the code the DLPR and it would have been duplicated here - I also thought this could be useful in other scenarios, where a library has a "standard" corporate DLPR bean in a jar and a user wants to use that, plus add more headers.